spec/gemstash/http_client_spec.rb

```diff
@@ -20,6 +20,16 @@
     @other_server.stop
   end
 
+  describe "#for" do
+    it "sets configured timeout value" do
+      configured_timeout = 99
+      client = Gemstash::HTTPClient.for(Gemstash::Upstream.new(@server.url), configured_timeout)
+      expect(
+        client.instance_variable_get(:@client).options.timeout
+      ).to eq(configured_timeout)
+    end
```

Comment:
@pcarranza FYI: It is going to be quite a hack to force the `SimpleServer` to sleep during the `get`. The problem with a "slow mount" method is that it will pause or `sleep` during the mount, not during the get. So it will merrily sleep for x number of seconds before defining the path you want to mount, then we finally get to the actual `get` call where we expect Faraday (NET::HTTP actually) to timeout. The sleep already happened further up the stack in the `slow_mount` call.

So in order to force a sleep during the `get` request, we need to break open `Servlet. The `Servlet` class gets passed to `WEBrick::HTTPServer.mount` which calls `initialize` on `Servlet` so we can't directly pass a timeout value to the servlet. We could define a `SlowServlet` class but then we have to parameterize it at the `SimpleServer` level by either making a `SlowSimpleServer` as well or passing an argument to `initialize`.

Hopefully all this makes sense. In other words it is a major PITA to try and setup the test to actually make a real `get` call and timeout using the passed fetch_timeout value. Not sure how you want to proceed. Let me know.